README.md

```diff
@@ -1,2 +1,154 @@
-# holonix
+# Holonix
+
 Holochain app development environment based on Nix.
+
+## Quickstart - Dev shell from default template
+
+hApp developers can use a Nix command to generate a flake that includes Holochain and other required binaries in a dev shell.
+
+```shell
+nix flake init -t github:holochain/holonix
+```
+
+You may be asked to allow substituters to be set to make use of binaries cached on cachix. This is not required to generate the flake, but when invoking the dev shell, Nix can benefit from the cache and download all binaries instead of building them on your machine from scratch.
+
+Inspecting the flake reveals the definition found under [./templates/default/flake.nix](./templates/default/flake.nix).
+
+The command to enter the dev shell and have the defined binaries available for execution is
+
+```shell
+nix develop
+```
+
+With the binary cache configured correctly, most or all binaries will be downloaded. How long this takes depends on your internet connection speed, but in any case the first time the dev shell is invoked will take longer than subsequent times.
+
+After the dev shell has built, Holochain and scaffolding commands are available.
+
+```shell
+[holonix:]$ holochain --version
+holochain x.y
+[holonix:]$ hc-scaffold --version
+holochain_scaffolding_cli x.y
+```
+
+## Updating Holonix flake
```

Comment:
We need to mention in this section that as new Holochain versions are released, we add them to the cache. Over time, older versions get removed from the cache. So if you don't stay near the most recent versions, you will end up with your CI rebuilding everything from scratch. 

At that point, you either need your own cachix cache or you need to update your inputs.